### ISSUE TITLE
Different result between models (stage 2) trained from scratch and provided checkpoint

### ISSUE THREAD

**Issue by hlinhn** [wei-mao-2019/ContAwareMotionPred#3]
Thank you for your work! I ran into some problems training the model (so far only stage 2). The provided checkpoint produced results close to the paper while running `eval_gta_stats` on GTA test set
```
path_err,57.38540468983287,94.38535354791149,118.40502109345802,152.13748948140577,87.92836402906703
joint_err,48.21895089142483,66.53363690807149,78.73921221957877,89.39613546094587,61.86808230991969
all_joint_err,73.7825790174371,111.85523976853166,136.15931872049035,167.86036358550845,103.99606321022947
```
but a model trained from scratch (stage 2 only) with GTA-IM data performed much worse
```
path_err,116.86713110841265,231.33667114892663,342.41806149832314,474.7360402887518,237.56161213853375
joint_err,172.96141145690794,256.80306212992963,337.8603548423286,437.9991502705907,253.4729992268008
all_joint_err,208.02058091611107,346.54184607816234,479.84686729845066,638.2465369540575,346.61500541655346
```
This is from the config file provided, with the number of epochs set to 50, the same as the paper. I have also tried
- increasing the number of epochs to 100
- changing the config file to match the numbers in the paper (specifically learning rate, number of DCT components, lambdas)
but they always performed around the same. 

What am I missing? 
In case it is a data problem: I downloaded GTA-IM FPS-30 dataset, processed it with `process_gta_dataset` and got 329 files in `data_v2_downsample0.02` folder. 

Thank you for your help!

**Comment by firrice** [comment]
Same problem during reproducing results, and i even couldn't reproduce the similar results using the provided checkpoints solely, which were depicted as below:

<img width="480" height="72" alt="Image" src="https://github.com/user-attachments/assets/0b9a80c2-917a-40ca-8318-2f272da0746c" />

By the way, same implementation details as "I downloaded GTA-IM FPS-30 dataset, processed it with process_gta_dataset and got 329 files in data_v2_downsample0.02 folder." etc.

which evaluation command have u used to make your result below?

<img width="999" height="177" alt="Image" src="https://github.com/user-attachments/assets/bb2f8bf5-7b60-4de5-8fd0-7616ca6e79d9" />

I have tried both two commands within the run.sh, however it all turned out the wrong results.  

<img width="985" height="50" alt="Image" src="https://github.com/user-attachments/assets/692e4495-4471-4c3a-8671-1273d9020a7b" />

Thanks u for your help!

**Comment by firrice** [comment]
I thought i have found reason, which there was a snippet of code in raw project need to annotated ┓( ´∀` )┏, shown as below:

<img width="714" height="646" alt="Image" src="https://github.com/user-attachments/assets/e6a6bfa2-d7ca-468f-839e-193830f9392b" />

And when the snippet, the results turn out:

<img width="724" height="58" alt="Image" src="https://github.com/user-attachments/assets/d8d41dfe-706b-43f1-8c37-0d750488f25e" />

However, it was still worse than that beyond to the original paper.
